# Upstream patch to fix a deprecation warning using AVStream.codec.time_base as muxer hint.
# URL: https://github.com/mltframework/mlt/commit/24a26b0099600a2b3cedfd1c627bf5d48379fdbc
# Closes: #760773

diff -Naur mlt-0.9.2.orig/src/modules/avformat/consumer_avformat.c mlt-0.9.2/src/modules/avformat/consumer_avformat.c
--- mlt-0.9.2.orig/src/modules/avformat/consumer_avformat.c	2014-09-24 19:04:25.073991603 +0200
+++ mlt-0.9.2/src/modules/avformat/consumer_avformat.c	2014-09-24 19:10:32.481981907 +0200
@@ -637,7 +637,11 @@
 
 		// Set parameters controlled by MLT
 		c->sample_rate = mlt_properties_get_int( properties, "frequency" );
+#if LIBAVFORMAT_VERSION_INT >= ((55<<16)+(44<<8)+0)
+		st->time_base = ( AVRational ){ 1, c->sample_rate };
+#else
 		c->time_base = ( AVRational ){ 1, c->sample_rate };
+#endif
 		c->channels = channels;
 
 		if ( mlt_properties_get( properties, "alang" ) != NULL )
@@ -837,8 +841,10 @@
 		c->height = mlt_properties_get_int( properties, "height" );
 		c->time_base.num = mlt_properties_get_int( properties, "frame_rate_den" );
 		c->time_base.den = mlt_properties_get_int( properties, "frame_rate_num" );
+#if LIBAVFORMAT_VERSION_INT < ((55<<16)+(44<<8)+0)
 		if ( st->time_base.den == 0 )
-			st->time_base = c->time_base;
+#endif
+		st->time_base = c->time_base;
 		c->pix_fmt = pix_fmt ? av_get_pix_fmt( pix_fmt ) : PIX_FMT_YUV420P;
 		
 		switch ( colorspace )
